core/templates/components/shared-component.module.ts

```diff
@@ -400,96 +396,93 @@ import { UnsavedChangesStatusInfoModalComponent } from './unsaved-changes-status
     SharedFormsModule,
     StringUtilityPipesModule,
     // Components, directives, and pipes.
+    AddOrUpdateSolutionModalComponent,
     AttributionGuideComponent,
     AudioBarComponent,
     AudioFileUploaderComponent,
+    CollectionSummaryTileComponent,
     CompletionGraphComponent,
-    CorrectnessFooterComponent,
     ConfirmLeaveModalComponent,
     ConfirmQuestionExitModalComponent,
-    ContinueButtonComponent,
     ContentLanguageSelectorComponent,
+    ContinueButtonComponent,
     ConversationSkinComponent,
-    EndChapterCheckMarkComponent,
-    EndChapterConfettiComponent,
     ConversationSkinEmbedComponent,
-    CreateNewSkillModalComponent,
+    CorrectnessFooterComponent,
     CreateActivityModalComponent,
+    CreateNewSkillModalComponent,
     CustomizeInteractionModalComponent,
     DeleteAnswerGroupModalComponent,
     DeleteHintModalComponent,
     DeleteInteractionModalComponent,
     DeleteLastHintModalComponent,
     DeleteSolutionModalComponent,
-    DisplaySolutionModalComponent,
-    DisplaySolutionInterstititalModalComponent,
     DisplayHintModalComponent,
+    DisplaySolutionInterstititalModalComponent,
+    DisplaySolutionModalComponent,
+    EditThumbnailModalComponent,
+    EndChapterCheckMarkComponent,
+    EndChapterConfettiComponent,
     ExplorationFooterComponent,
     ExplorationSummaryTileComponent,
-    FeedbackPopupComponent,
-    LearnerAnswerInfoCard,
-    MultiSelectionFieldComponent,
     FilteredChoicesFieldComponent,
-    PracticeTabComponent,
-    CollectionSummaryTileComponent,
+    FeedbackPopupComponent,
     HintAndSolutionButtonsComponent,
     HintEditorComponent,
     InputResponsePairComponent,
     LazyLoadingComponent,
-    ProfileLinkImageComponent,
-    PreviewThumbnailComponent,
-    RatingDisplayComponent,
-    RatingsAndRecommendationsComponent,
-    ResponseHeaderComponent,
-    RubricsEditorComponent,
+    LearnerAnswerInfoCard,
+    LearnerDashboardIconsComponent,
+    MultiSelectionFieldComponent,
     OnScreenKeyboardComponent,
     OutcomeDestinationEditorComponent,
     OutcomeEditorComponent,
     OutcomeFeedbackEditorComponent,
+    PostChapterRecommendationsComponent,
```

Comment:
Why is this component added here? I see that the file is present in pages/exploration-player-page. Let's not pollute the shared components with files not in the components folder.

Reply:
Actually, when adding this component to the exploration player page module, it results in the following console error:

```
Unhandled Promise rejection: Template parse errors:
Can't bind to 'nextStoryNodeLink' since it isn't a known property of 'oppia-post-chapter-recommendations'.
1. If 'oppia-post-chapter-recommendations' is an Angular component and it has 'nextStoryNodeLink' input, then verify that it is part of this module.
2. If 'oppia-post-chapter-recommendations' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component. ("f="inStoryMode && isEndChapterFeatureEnabled() && storyId">
    <oppia-post-chapter-recommendations [ERROR ->][nextStoryNodeLink]="nextLessonLink" [nextStoryNodeThumbnailUrl]="nextStoryNodeIconUrl ? nextStoryNod"): ng:///SharedComponentsModule/RatingsAndRecommendationsComponent.html@45:40
```

`PostChapterRecommendationsComponent` is used inside the ratings-and-recommendations component, which is registered in the shared-components module. I'm not sure, but maybe that's the reason adding the`PostChapterRecommendationsComponent` to the exploration player page module causes the error?

Reply:
From the offline discussion:

PostChapterRecommendationsComponent is used in ratings and recommendation. This component is used in a lot of pages. So we have decided to create a new module called recommendations in the core/templates/components folder for the recommendations related code.

Reply:
I've moved the new `PostChapterRecommendationsComponent` into a separate folder in the components directory, and created a new module to register the component in and finally imported this module into the shared-components module.

However, I have two queries.

 - I also tried moving the `ratings-and-recommendations` component like we discussed, but I faced an issue. The component uses the `exploration-summary-tile` component in its template. This requirement was initially satisfied because both the components were in the `shared-components` module. However, after moving the `ratings-and-recommendations` component into its own module, the summary tile component now needs to be provided in this module as well for the functioning of the `ratings-and-recommendations` component.
 
 -- Since declaring the `summary-tile` component it in two places is not an option, I think the ideal solution is to package all summary tile components (`exploration-summary-tile`, `collection-summary-tile`, etc.) into a discrete module and use this module instead. However, that could take a while and I unfortunately do not have the bandwidth to do so at the moment; I may be able to take up this issue at a later date.
 
  - Also, I need to add codeowners for the newly created recommendations folder in the components directory. Who should I be adding as a codeowner here?

Reply:
Approved. Can you create an issue so that it is being tracked?